jetty-infinispan/infinispan-common/src/main/java/org/eclipse/jetty/session/infinispan/SessionDataMarshaller.java

```diff
@@ -39,35 +38,24 @@
  * control to ensure that session attributes can be deserialized using either
  * the container class loader or the webapp classloader, as appropriate.
  */
-public class SessionDataMarshaller implements MessageMarshaller<InfinispanSessionData>, Externalizer<InfinispanSessionData>
+public class SessionDataMarshaller
+        implements MessageMarshaller<InfinispanSessionData>, Externalizer<InfinispanSessionData>
 {
     /**
      * The version of the serializer.
      */
     private static final int VERSION = 0;
 
-    private static final Logger LOG = Log.getLogger("org.eclipse.jetty.server.session");
-    
-    private static SerializationContext serializationContext;
-
-    static
+    private SerializationContext initSerializationContext() throws IOException
     {
         FileDescriptorSource fds = new FileDescriptorSource();
```

Comment:
It is useless to keep SerializationContext because a new SessionDataMarshaller instance is created every time readObject or writeObject is called. I can either have it as a lazily initialized static field or create one every time

Reply:
Lets make it a static, lazily initialized. That way, if the remote protobuf marshalling is being used, it will never be initialized, it is only if the Externalizer methods are called that it will be initialized.